src/ducks/sep24WithdrawAsset.ts

```diff
@@ -11,48 +11,70 @@ import {
   sep10AuthSend,
 } from "methods/sep10Auth";
 import {
-  checkToml,
   checkInfo,
   interactiveWithdrawFlow,
   createPopup,
   pollWithdrawUntilComplete,
 } from "methods/sep24";
+import { checkTomlForSep } from "methods/checkTomlForSep";
 import {
+  Asset,
   ActionStatus,
   RejectMessage,
   Sep24WithdrawAssetInitialState,
+  TomlFields,
+  CheckInfoType,
 } from "types/types.d";
 
 export const withdrawAssetAction = createAsyncThunk<
   { currentStatus: string },
-  { assetCode: string; assetIssuer: string },
+  Asset,
   { rejectValue: RejectMessage; state: RootState }
 >(
   "sep24WithdrawAsset/withdrawAssetAction",
-  async ({ assetCode, assetIssuer }, { rejectWithValue, getState }) => {
+  async (asset, { rejectWithValue, getState }) => {
+    const { assetIssuer, assetCode, homeDomain } = asset;
     const { data, secretKey } = accountSelector(getState());
     const { pubnet } = settingsSelector(getState());
     const networkConfig = getNetworkConfig(pubnet);
     const publicKey = data?.id;
 
+    // This is unlikely
     if (!publicKey) {
       throw new Error("Something is wrong with Account, no public key.");
     }
 
+    // This is unlikely
+    if (!homeDomain) {
+      throw new Error("Something went wrong, home domain is not defined.");
+    }
+
     try {
       // Check toml
-      const tomlResponse = await checkToml({
+      const tomlResponse = await checkTomlForSep({
+        sepName: "SEP-24 withdrawal",
         assetIssuer,
+        requiredKeys: [
+          TomlFields.SIGNING_KEY,
+          TomlFields.TRANSFER_SERVER_SEP0024,
+          TomlFields.WEB_AUTH_ENDPOINT,
+        ],
         networkUrl: networkConfig.url,
+        homeDomain,
       });
 
       // Check info
-      await checkInfo({ toml: tomlResponse, assetCode });
+      await checkInfo({
+        type: CheckInfoType.WITHDRAWAL,
+        toml: tomlResponse,
+        assetCode,
+      });
 
       // SEP-10 start
       const challengeTransaction = await sep10AuthStart({
         authEndpoint: tomlResponse.WEB_AUTH_ENDPOINT,
-        secretKey,
+        publicKey,
+        homeDomain,
```

Comment:
Doesn't necessarily have to be in this PR, but there's a fair amount of repetition between the withdraw and deposit flow. Might be nice to wrap all the checking and signing stuff into a helper function